lib/matplotlib/gridspec.py

```diff
@@ -627,7 +627,6 @@ def _from_subplot_args(figure, args):
             if not (isinstance(rows, Integral) and isinstance(cols, Integral)):
                 _api.warn_deprecated("3.3", message=message)
                 rows, cols = map(int, [rows, cols])
-            gs = GridSpec(rows, cols, figure=figure)
```

Comment:
So I guess this had a side effect of raising when `rows` or `cols` were invalid, before the check for `num`; I wonder if we should copy that check here, or leave this in here without assignment. Does this have any other side effect if input is _valid_?

Reply:
I think it's fine to have this go away as the same checks will be done by check_gridspec_exists (which would return False if there's no such gridspec) or the GridSpec constructor at the end.

Reply:
Well, the tests fail because this early check used to say something about `rows`/`cols`, but now it says that `num` is invalid (because it's greater than 0 as used in the test.) The latter check/constructor does not even see it.

Reply:
I don't see any reason you can't move the actual gs code from below up to here?  i, j and num aren't needed until the return...

Reply:
I simplified this code as part of #20447, which removed the redundant assignment, so it's no longer in this PR.